src/typedefs.jl

```diff
@@ -6,18 +6,20 @@ const DefaultRefType = UInt32
 # This workaround is needed since this type not defined yet
 # See JuliaLang/julia#269
 mutable struct CategoricalPool{T, R <: Integer, V}
-    index::Vector{T}
-    invindex::Dict{T, R}
-    order::Vector{R}
-    levels::Vector{T}
-    valindex::Vector{V}
+    index::Vector{T}        # category levels ordered by their reference codes
+    invindex::Dict{T, R}    # map from category levels to their reference codes
+    order::Vector{R}        # 1-to-1 map from `index` to `level` (position of i-th category in `levels`)
+    levels::Vector{T}       # category levels ordered by externally specified order
+    valindex::Vector{V}     # "category value" objects 1-to-1 matching `index`
     ordered::Bool
 
     function CategoricalPool{T, R, V}(index::Vector{T},
                                       invindex::Dict{T, R},
                                       order::Vector{R},
                                       ordered::Bool) where {T, R, V}
-        pool = new(index, invindex, order, index[order], V[], ordered)
+        levels = similar(index)
+        levels[order] = index # could use ipermute!(), but this should be faster
```

Comment:
Don't leave this "should" here: either it's faster or it's not, comments should be factual.

Reply:
You're right.I just wanted a reminder how these order is an inverse permutation from index to levels